File: store/src/main/java/com/nytimes/android/external/store3/base/DiskAllRead.java
package com.nytimes.android.external.store3.base;


import java.io.FileNotFoundException;

import javax.annotation.Nonnull;

import io.reactivex.Observable;

public interface DiskAllRead<Raw> {

    /**
     * Use {@link #safeReadAll(String)} instead
     */
    @Deprecated
    @Nonnull
    Observable<Raw> readAll(@Nonnull String path) throws FileNotFoundException;

    @Nonnull
    Observable<ReadResult<Raw>> safeReadAll(@Nonnull final String path);
}

File: store/src/main/java/com/nytimes/android/external/store3/base/DiskAllRead.kt
package com.nytimes.android.external.store3.base


import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.channels.ReceiveChannel
import java.io.FileNotFoundException

interface DiskAllRead<Raw> {
    @Throws(FileNotFoundException::class)
    suspend fun CoroutineScope.readAll(path: String): ReceiveChannel<Raw>
}

